btsieve/src/ethereum/web3_http_blocksource.rs

```diff
@@ -61,7 +71,70 @@ impl BlockSource for Web3HttpBlockSource {
                             _ => Err(error)
                         }
                     })
-                    .map_err(Error::Source)
+                    .map_err(Error::Web3)
+            })
+            .filter_map(|maybe_block| maybe_block)
```

Comment:
I think it's because the previous combinator returns `Result<Option<Block>,Error>`. This will take all the `Option<Block>`s and only yield the `Some(Block)`s, filtering out the `None`s.